program/src/processor.rs

```diff
@@ -0,0 +1,59 @@
+use {
+    crate::instruction::SolStakeViewInstruction,
+    num_traits::FromPrimitive,
+    solana_program::{
+        account_info::{next_account_info, AccountInfo}, clock::Clock, borsh1::try_from_slice_unchecked, entrypoint::ProgramResult, msg, program_error::ProgramError, pubkey::Pubkey, stake, sysvar::{self, Sysvar}, program::set_return_data,
+    }
+};
+
+pub fn process_instruction<'a>(
+    _program_id: &Pubkey,
+    accounts: &'a [AccountInfo<'a>],
+    instruction_data: &[u8],
+) -> ProgramResult {
+    if instruction_data.is_empty() {
+        return Err(ProgramError::InvalidArgument);
+    }
+    let instruction = SolStakeViewInstruction::from_u8(instruction_data[0]).ok_or(ProgramError::InvalidArgument)?;
+    match instruction {
+        SolStakeViewInstruction::GetStakeActivatingAndDeactivating => {
+            msg!("Instruction: GetStakeActivatingAndDeactivating");
+            get_stake_activating_and_deactivating(accounts)
+        }
+    }
+}
+
+const RETURN_DATA_SIZE: usize = 56;
```

Comment:
Can we add a comment here with the layout of the return data perhaps? Something like:
- `[ 0..32]`: validator vote (`Pubkey`)
- `[32..40]`: effective stake (`u64`)
-  `[40..48]`: activating stake (`u64`)
-  `[48..56]`:  deativating stake (`u64`)

If there is no delegation, return data will be zeroed.